script.js

```diff
@@ -1,4 +1,346 @@
+    myOutputValue = myOutputValue + "<br><br>" + leaderboardText;
+    +`<br><br> You can resubmit the number of dice that you want to play with!<br><br>`;
+    mode = "numberOfDice";
+    return myOutputValue;
+  }
+  if (knockoutMode == false) {
+    var leadingScore = currentRoundCombinedNums[0];
+    var index = 0;
+    var winnerText = "";
+    while (index < currentRoundCombinedNums.length) {
+      if (isNormalMode == true) {
+        if (currentRoundCombinedNums[index] > leadingScore) {
+          leadingScore = currentRoundCombinedNums[index];
+          winnerText = `The winner is Player ${
+            index + 1
+          } who rolled the highest value of ${leadingScore}`;
+        } else if (leadingScore == currentRoundCombinedNums[0]) {
+          winnerText = `The winner is Player 1 who rolled the highest value of ${leadingScore}`;
+        }
+      } else if (isNormalMode == false) {
+        if (currentRoundCombinedNums[index] < leadingScore) {
+          leadingScore = currentRoundCombinedNums[index];
+          winnerText = `The winner is Player ${
+            index + 1
+          } who rolled the lowest value of ${leadingScore}`;
+        } else if (leadingScore == currentRoundCombinedNums[0]) {
+          winnerText = `The winner is Player 1 who rolled the lowest value of ${leadingScore}. <br><br> You can input the number of dice you wish to play with next round!`;
+        }
+      }
+      index = index + 1;
+    }
+    var leaderboardText = leaderboard(leaderboardArray, currentWinner);
+    myOutputValue = winnerText + `<br><br>` + leaderboardText;
+    return myOutputValue;
+  }
+};
+
+var leaderboard = function (leaderboardArray, currentWinner) {
+  leaderboardArray[currentWinner - 1] = leaderboardArray[currentWinner - 1] + 1;
+  var index = 0;
+  var myOutputValue = "";
+  while (index < noOfPlayers) {
+    myOutputValue =
+      myOutputValue +
+      `Player ${index + 1} has won a total of: ${
+        leaderboardArray[index]
+      } rounds. <br><br>`;
+    index = index + 1;
+  }
   return myOutputValue;
 };
+
+var inputDisplay = function () {
+  if (mode == "numberOfPlayers") {
+    return "Please input the number of Players for this round.";
+  }
+  if (mode == "numberOfDice") {
+    return "Please input the number of Dice in play for this round.";
+  }
+  if (mode == "autoModeSelect") {
+    return "Please input 'yes' or 'no' if you wish to have rolls automatically combined.";
+  }
+  if (mode == "knockoutModeSelect") {
+    return "Please input 'yes' or 'no' if you wish to have a round knockout system.";
+  }
+  if (mode == "gameModeSelect") {
+    return "Please input normal or lowest to select which mode to play.";
+  }
+  if (mode == "gameMode") {
+    return `Please submit to roll the dice.`;
+  }
+  if (mode == "selectMode") {
+    return `Player ${
+      currentPlayer + 1
+    }, please select your die in order starting from 0.`;
+  } else if (mode == "faceOff") {
+    return "The game has ended!";
```

Comment:
since you have multiple cases, you could consider using a switch statement instead (https://www.w3schools.com/js/js_switch.asp). However good job using if statements to implement this logic